File: app/src/main/java/com/example/josycom/flowoverstack/repository/QuestionRepository.java
package com.example.josycom.flowoverstack.repository;

import androidx.lifecycle.LiveData;
import androidx.paging.LivePagedListBuilder;
import androidx.paging.PagedList;

import com.example.josycom.flowoverstack.model.Question;
import com.example.josycom.flowoverstack.model.QuestionDataSourceFactory;

public class QuestionRepository {
    private int page;
    private int pageSize;
    private String order;
    private String sortCondition;
    private String site;
    private String filter;
    private LiveData<PagedList<Question>> mQuestionPagedList;

    public QuestionRepository(int page, int pageSize, String order, String sortCondition, String site, String filter) {
        this.page = page;
        this.pageSize = pageSize;
        this.order = order;
        this.sortCondition = sortCondition;
        this.site = site;
        this.filter = filter;
        init();
    }

    private void init() {
        QuestionDataSourceFactory factory = new QuestionDataSourceFactory(page, pageSize, order, sortCondition, site, filter);
        PagedList.Config pageConfig = new PagedList.Config.Builder()
                .setEnablePlaceholders(false)
                .setPageSize(pageSize)
                .build();
        mQuestionPagedList = new LivePagedListBuilder<>(factory, pageConfig).build();
    }

    public LiveData<PagedList<Question>> getQuestionPagedList() {
        return mQuestionPagedList;
    }
}

File: app/src/main/java/com/josycom/mayorjay/flowoverstack/data/repository/QuestionRepository.kt
package com.josycom.mayorjay.flowoverstack.data.repository

import androidx.paging.Pager
import androidx.paging.PagingConfig
import androidx.paging.PagingData
import com.josycom.mayorjay.flowoverstack.data.model.Question
import com.josycom.mayorjay.flowoverstack.data.remote.datasource.QuestionPagingSource
import com.josycom.mayorjay.flowoverstack.data.remote.service.ApiService
import kotlinx.coroutines.flow.Flow
import javax.inject.Inject
import javax.inject.Singleton

@Singleton
class QuestionRepository @Inject constructor(private val apiService: ApiService) {

    var questionDataFlow: Flow<PagingData<Question>>? = null

    fun init(page: Int, pageSize: Int, order: String, sortCondition: String, site: String, tagged: String, filter: String, siteKey: String) {
        questionDataFlow = Pager(PagingConfig(pageSize, enablePlaceholders = false),
                pagingSourceFactory = { QuestionPagingSource(page, pageSize, order, sortCondition, site, tagged, filter, siteKey, apiService) }).flow
    }
}